physical/metadata/metadata.go

```diff
@@ -40,9 +41,124 @@ func (meta *NodeMetadata) EventTimeField() octosql.VariableName {
 	return meta.eventTimeField
 }
 
-func NewNodeMetadata(cardinality Cardinality, eventTimeField octosql.VariableName) *NodeMetadata {
+func (meta *NodeMetadata) Namespace() *Namespace {
+	return meta.namespace
+}
+
+func NewNodeMetadata(cardinality Cardinality, eventTimeField octosql.VariableName, namespace *Namespace) *NodeMetadata {
 	return &NodeMetadata{
 		cardinality:    cardinality,
 		eventTimeField: eventTimeField,
+		namespace:      namespace,
+	}
+}
+
+func NewNodeMetadataFromMetadata(meta *NodeMetadata) *NodeMetadata {
+	return &NodeMetadata{
+		cardinality:    meta.Cardinality(),
+		eventTimeField: meta.EventTimeField(),
+		namespace:      meta.Namespace(),
+	}
+}
+
+type Namespace struct {
+	prefixes []string
+	names    []octosql.VariableName
```

Comment:
I think this unnecessary, and doesn't contain the full info anyways (we can't know the contents of a table for example), so I think you should just get the prefixes of variables in the constructor and save those.

Reply:
I don't think I understood what you mean - you want to just store prefixes from datasources (and only change them on requalifiers) and don't bother with single names as in a map?

Reply:
Yes, indeed. Just the datasource aliases really.